SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestRemotePossession.cpp

```diff
@@ -31,25 +32,84 @@ void ASpatialTestRemotePossession::PrepareTest()
 		FinishStep();
 	});
 
-	AddStep(TEXT("Create Pawn"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float DeltaTime) {
+	AddStep(TEXT("Create Pawn"), FWorkerDefinition::Server(1), nullptr, [this]() {
 		ATestPossessionPawn* Pawn =
 			GetWorld()->SpawnActor<ATestPossessionPawn>(LocationOfPawn, FRotator::ZeroRotator, FActorSpawnParameters());
 		RegisterAutoDestroyActor(Pawn);
 		FinishStep();
 	});
 
+	AddStep(TEXT("Clear Original Pawn Array"), FWorkerDefinition::AllServers, nullptr, [this]() {
+		OriginalPawns.Empty();
+		FinishStep();
+	});
+
 	// Ensure that all Controllers are located on the right Worker
 	AddWaitStep(FWorkerDefinition::AllServers);
 
+	AddStep(TEXT("Save Original Pawns"), FWorkerDefinition::AllServers, nullptr, [this]() {
+		for (ASpatialFunctionalTestFlowController* FlowController : GetFlowControllers())
+		{
+			ATestPossessionPlayerController* PlayerController = Cast<ATestPossessionPlayerController>(FlowController->GetOwner());
+			if (PlayerController != nullptr && PlayerController->HasAuthority() && PlayerController->GetPawn()->HasAuthority())
+			{
+				AddToOriginalPawns(PlayerController, PlayerController->GetPawn());
+				UE_LOG(LogTemp, Log, TEXT("Adding original pawn %s to array"), *PlayerController->GetPawn()->GetName());
+			}
+		}
+		FinishStep();
+	});
+
 	ATestPossessionPlayerController::ResetCalledCounter();
 }
 
+void ASpatialTestRemotePossession::AddCleanupSteps()
+{
+	AddStep(TEXT("Clean up the test"), FWorkerDefinition::AllServers, nullptr, nullptr, [this](float) {
+		for (const auto& OriginalPawnPair : OriginalPawns)
+		{
+			if (OriginalPawnPair.Controller != nullptr && OriginalPawnPair.Controller->HasAuthority())
+			{
+				OriginalPawnPair.Controller->UnPossess();
+				OriginalPawnPair.Controller->RemotePossessOnServer(OriginalPawnPair.Pawn);
+				AssertIsValid(OriginalPawnPair.Pawn, TEXT("Original pawn shouldn't be null"));
```

Comment:
This feels a bit weird, this is testing the assumptions the test itself makes rather than the tested scenario if that makes sense? If we want to be defensive here we should use a normal unreal assert: `check()`

Reply:
This was part of our temp debugging and I forgot to remove it 😅